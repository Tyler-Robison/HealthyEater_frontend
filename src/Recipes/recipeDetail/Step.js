
/** Step components are items in the StepsList component
 * 
 * Each step contains a step number, instructions and equipment/ingredients needed*/
const Step = ({ step }) => {
    const displayItems = (step) => {
        if (step.length === 0) return 'None'

        const itemArr = step.map(item => item.name)
        return itemArr.join(', ')
    }

    return (
        <div className='step-div col-10'>

            <p className='step-number'><b>{step.number}</b></p>
            <p>{step.step}</p>

            <p><b>Equipment Used:</b> {displayItems(step.equipment)} </p>
            <p><b>Ingredients Used:</b> {displayItems(step.ingredients)} </p>

        </div>
    )
}

export default Step;