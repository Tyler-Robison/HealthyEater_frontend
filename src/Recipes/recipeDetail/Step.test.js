import React from 'react';
import { render, screen } from '@testing-library/react';
import Step from './Step';
import { MemoryRouter } from "react-router-dom";
import ContextProvider from '../../testContext';

const step = {
    number: 1,
    step: 'Do the first step',
    equipment: [{ name: 'mock equipment' }, { name: 'other equipment' }],
    ingredients: [{ name: 'mock ingredient' }, { name: 'other ingredient' }]
}

it("renders without crashing", function () {
    <MemoryRouter>
        <ContextProvider >
            <Step step={step} />
        </ContextProvider>
    </MemoryRouter>
});

it("matches snapshot.", function () {

    const container = render(
        <MemoryRouter>
            <ContextProvider >
                <Step step={step} />
            </ContextProvider>
        </MemoryRouter>
    );
    expect(container.asFragment()).toMatchSnapshot();
});