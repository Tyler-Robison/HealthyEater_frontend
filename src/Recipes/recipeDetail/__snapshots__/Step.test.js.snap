// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot. 1`] = `
<DocumentFragment>
  <div
    class="step-div"
  >
    <p
      class="step-number"
    >
      <b>
        1
      </b>
    </p>
    <p>
      Do the first step
    </p>
    <p>
      <b>
        Equipment Used:
      </b>
       mock equipment, other equipment 
    </p>
    <p>
      <b>
        Ingredients Used:
      </b>
       mock ingredient, other ingredient 
    </p>
  </div>
</DocumentFragment>
`;
